test/hooks/useLatestResult-test.tsx

```diff
@@ -41,50 +40,42 @@ describe("useLatestResult", () => {
             await sleep(180);
             return query;
         };
+        const { rerender } = render(<LatestResultsComponent query={0} doRequest={doRequest} />);
+        await act(() => sleep(100));
```

Comment:
Can we get rid of the sleeps? For instance by using `flushPromises` or `findBy` queries? Or if the tests require a specific timing fake timers.

Reply:
That was a good little task, have removed all the sleeps and the best way to do that was to refactor it to purely test the hook instead of using a dummy component. Let me know what you think - I'd say this counts as a rewrite of the test now as opposed to a refactor. I've expanded the tests to add a couple of smaller steps at the beginning, but then reproduced the original tests (in spirit).

Reply:
Marking as a rereview to account for the changes

Reply:
Looks really good. Thanks :+1: